### PR TITLE
Refactor `visitMaxs()` asm calls to handle errors 

### LINKED ISSUE #39043: [Improvement]: Add function name to the visitMaxs erros
### Description

Please wrap following line with a try-cache for runtime exceptions and warp it with our own error with function name (an location if possible).

https://github.com/ballerina-platform/ballerina-lang/blob/0fe9a823d9a7a270c92606484e48d14bdb81f3cd/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MethodGen.java#L303

This will help a lot when we get `visitMaxs` bugs, to report bugs and find work-arounds.

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject
Fixes #39043

## Approach
Moved the calls to a `Util` method and handle errors.

## Samples
If we remove some lines in the current code generation, it will throw an `asm` related exception at the `visitMaxs()` method call. 
```
java.lang.NullPointerException
        at org.objectweb.asm.Frame.merge(Frame.java:1171)
        at org.objectweb.asm.MethodWriter.computeAllFrames(MethodWriter.java:1610)
        at org.objectweb.asm.MethodWriter.visitMaxs(MethodWriter.java:1546)
        at org.wso2.ballerinalang.compiler.bir.codegen.methodgen.MethodGen.genJMethodForBFunc(MethodGen.java:303)
        at org.wso2.ballerinalang.compiler.bir.codegen.methodgen.MethodGen.generateMethod(MethodGen.java:173)
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.lambda$generateModuleClasses$0(JvmPackageGen.java:446)
        at java.base/java.util.HashMap$EntrySet.forEach(HashMap.java:1040)
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.generateModuleClasses(JvmPackageGen.java:399)
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.generate(JvmPackageGen.java:810)
        at org.wso2.ballerinalang.compiler.bir.codegen.CodeGenerator.generate(CodeGenerator.java:96)
        at org.wso2.ballerinalang.compiler.bir.codegen.CodeGenerator.generate(CodeGenerator.java:71)
        at io.ballerina.projects.JBallerinaBackend.performCodeGen(JBallerinaBackend.java:318)
        at io.ballerina.projects.ModuleContext.generateCodeInternal(ModuleContext.java:477)
        at io.ballerina.projects.ModuleCompilationState$4.generatePlatformSpecificCode(ModuleCompilationState.java:132)
        at io.ballerina.projects.ModuleContext.generatePlatformSpecificCode(ModuleContext.java:385)
        at io.ballerina.projects.JBallerinaBackend.performCodeGen(JBallerinaBackend.java:170)
        at io.ballerina.projects.JBallerinaBackend.<init>(JBallerinaBackend.java:139)
        at io.ballerina.projects.JBallerinaBackend.lambda$from$0(JBallerinaBackend.java:117)
        at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1134)
        at io.ballerina.projects.PackageCompilation.getCompilerBackend(PackageCompilation.java:170)
        at io.ballerina.projects.JBallerinaBackend.from(JBallerinaBackend.java:116)
        at org.ballerinalang.stdlib.utils.BuildLangLib.main(BuildLangLib.java:94)
Exception in thread "main" java.lang.NullPointerException
        at org.objectweb.asm.Frame.merge(Frame.java:1171)
        at org.objectweb.asm.MethodWriter.computeAllFrames(MethodWriter.java:1610)
        at org.objectweb.asm.MethodWriter.visitMaxs(MethodWriter.java:1546)
        at org.wso2.ballerinalang.compiler.bir.codegen.methodgen.MethodGen.genJMethodForBFunc(MethodGen.java:303)
        at org.wso2.ballerinalang.compiler.bir.codegen.methodgen.MethodGen.generateMethod(MethodGen.java:173)
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.lambda$generateModuleClasses$0(JvmPackageGen.java:446)
        at java.base/java.util.HashMap$EntrySet.forEach(HashMap.java:1040)
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.generateModuleClasses(JvmPackageGen.java:399)
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.generate(JvmPackageGen.java:810)
        at org.wso2.ballerinalang.compiler.bir.codegen.CodeGenerator.generate(CodeGenerator.java:96)
        at org.wso2.ballerinalang.compiler.bir.codegen.CodeGenerator.generate(CodeGenerator.java:71)
        at io.ballerina.projects.JBallerinaBackend.performCodeGen(JBallerinaBackend.java:318)
        at io.ballerina.projects.ModuleContext.generateCodeInternal(ModuleContext.java:477)
        at io.ballerina.projects.ModuleCompilationState$4.generatePlatformSpecificCode(ModuleCompilationState.java:132)
        at io.ballerina.projects.ModuleContext.generatePlatformSpecificCode(ModuleContext.java:385)
        at io.ballerina.projects.JBallerinaBackend.performCodeGen(JBallerinaBackend.java:170)
        at io.ballerina.projects.JBallerinaBackend.<init>(JBallerinaBackend.java:139)
        at io.ballerina.projects.JBallerinaBackend.lambda$from$0(JBallerinaBackend.java:117)
        at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1134)
        at io.ballerina.projects.PackageCompilation.getCompilerBackend(PackageCompilation.java:170)
        at io.ballerina.projects.JBallerinaBackend.from(JBallerinaBackend.java:116)
        at org.ballerinalang.stdlib.utils.BuildLangLib.main(BuildLangLib.java:94)
```

## CheckList 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCodeGenUtil.java [hunk 1]
```diff
@@ -828,4 +829,14 @@ private JvmCodeGenUtil() {
...
         return JvmConstants.TYPEREF_TYPE_VAR_PREFIX + Utils.encodeNonFunctionIdentifier(type.tsymbol.name.value);
     }
+
+    public static void visitMaxStackForMethod(MethodVisitor mv, String funcName, String className) {
+        try {
+            mv.visitMaxs(0, 0);
+        } catch (RuntimeException e) {
+            throw new BLangCompilerException(
+                    "error while generating method '" + Utils.decodeIdentifier(funcName) + "' in class '" +
+                            Utils.decodeIdentifier(className) + "'", e);
```
Comment (Nadeeshan96) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCodeGenUtil.java:839]: Shall we update the PR description to show how this error is displayed now?

**Comment by HindujaB** [comment]
The above error will now be returning the following stack trace. 
```
org.ballerinalang.compiler.BLangCompilerException: error while generating method 'ArgsData$gen$.<init>' in class 'ballerina/lang.annotations/0/$value$ArgsData'
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmCodeGenUtil.visitMaxStackForMethod(JvmCodeGenUtil.java:839)
        at org.wso2.ballerinalang.compiler.bir.codegen.methodgen.MethodGen.genJMethodForBFunc(MethodGen.java:304)
        at org.wso2.ballerinalang.compiler.bir.codegen.methodgen.MethodGen.generateMethod(MethodGen.java:174)
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmValueGen.createRecordMethods(JvmValueGen.java:414)
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmValueGen.createRecordValueClass(JvmValueGen.java:378)
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmValueGen.lambda$generateValueClasses$0(JvmValueGen.java:235)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmValueGen.generateValueClasses(JvmValueGen.java:220)
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.generate(JvmPackageGen.java:804)
        at org.wso2.ballerinalang.compiler.bir.codegen.CodeGenerator.generate(CodeGenerator.java:96)
        at org.wso2.ballerinalang.compiler.bir.codegen.CodeGenerator.generate(CodeGenerator.java:71)
        at io.ballerina.projects.JBallerinaBackend.performCodeGen(JBallerinaBackend.java:318)
        at io.ballerina.projects.ModuleContext.generateCodeInternal(ModuleContext.java:477)
        at
...
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCodeGenUtil.java (hunk 1 above)
Comment (HindujaB) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCodeGenUtil.java:839]: added.

**Review by warunalakshitha — APPROVED** [review]
LGTM

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCodeGenUtil.java [hunk 2]
```diff
@@ -828,4 +829,14 @@ private JvmCodeGenUtil() {
     public static String getRefTypeConstantName(BTypeReferenceType type) {
         return JvmConstants.TYPEREF_TYPE_VAR_PREFIX + Utils.encodeNonFunctionIdentifier(type.tsymbol.name.value);
     }
+
+    public static void visitMaxStackForMethod(MethodVisitor mv, String funcName, String className) {
+        try {
+            mv.visitMaxs(0, 0);
+        } catch (RuntimeException e) {
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCodeGenUtil.java:836]: Should we catch throwable here?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmConstants.java [hunk 3]
```diff
@@ -406,7 +408,10 @@ public class JvmConstants {
     public static final String OBSERVABLE_ANNOTATION = "ballerina/observe/Observable";
     public static final String DISPLAY_ANNOTATION = "display";
     public static final String RECORD_CHECKPOINT_METHOD = "recordCheckpoint";
-
+    public static final String CREATE_INTEROP_ERROR_METHOD = "createInteropError";
+    public static final String LAMBDA_PREFIX = "$lambda$";
+    public static final String POPULATE_METHOD_PREFIX = "$populate";
+    public static final String ADD_METHOD = "add";
```
Comment (Nadeeshan96) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmConstants.java:414]: These are not specific to `// observability related constants` right? Shall we move these under `// code generation related constants`

Comment (Nadeeshan96) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmConstants.java:414]: Other changes LGTM.

**Review by Nadeeshan96 — APPROVED** [review]
LGTM

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**